### PR TITLE
Object model fixes

### PR DESCRIPTION
**Pull request by barcharcraz**
Change az_error in the object model to az_result.

### CODE REVIEW

File: docs/clang/design.md [hunk 1]
```diff
@@ -788,10 +788,10 @@ To define a method on an object simply define a function taking a pointer to tha
  * @param[in] __[transfer none]__ cat - the cat to add
  * @return Any errors
  * @retval AZ_SUCCESS on success
- * @retval AZ_ERROR_NO_MEMORY if a reallocation of the internal
+ * @retval AZ_RESULT_NO_MEMORY if a reallocation of the internal
```
Comment (sergey-shandar) [inline docs/clang/design.md]: We still use `AZ_ERROR_...` in cases of error so it will be `AZ_ERROR_OUT_OF_MEMORY`.

File: docs/clang/design.md [hunk 2]
```diff
@@ -788,10 +788,10 @@ To define a method on an object simply define a function taking a pointer to tha
  * @param[in] __[transfer none]__ cat - the cat to add
  * @return Any errors
  * @retval AZ_SUCCESS on success
```
Comment (sergey-shandar) [inline docs/clang/design.md]: `AZ_OK`. But users shouldn't use `result == AZ_OK`. they should use an inline function `az_sucesses`

Comment (barcharcraz) [reply docs/clang/design.md]: `az_success`?

**Comment by adrianhall** [comment]
@sergey-shandar @barcharcraz - where are we on this one?  Looks like a couple of minor changes have been checked in and it is good to go?

File: docs/clang/design.md (hunk 2 above)
Comment (sergey-shandar) [reply docs/clang/design.md]: https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/core/core/inc/az_result.h

Comment (sergey-shandar) [reply docs/clang/design.md]: `az_succeeded`

**Pull request merged**